include/bits/exec.h

```diff
@@ -0,0 +1,74 @@
+#include <string.h>
+
+int execl(const char *path, const char* argv0, ...)
+{
+	int argc;
+	va_list ap;
+	va_start(ap, argv0);
+	for (argc=1; va_arg(ap, const char*); argc++);
+	va_end(ap);
+	{
+		int i;
+		char *argv[argc+1];
+		va_start(ap, argv0);
+		argv[0] = (char *)argv0;
+		for (i=1; i<argc; i++)
+			argv[i] = va_arg(ap, char *);
+		argv[i] = NULL;
+		va_end(ap);
+		return _execl(path, argv);
+	}
+}
+
+int execle(const char *path, const char* argv0, ...)
+{
+	int argc;
+	va_list ap;
+	va_start(ap, argv0);
+	for (argc=1; va_arg(ap, const char *); argc++);
+	va_end(ap);
+	{
+		int i;
+		char *argv[argc+1];
+		char **envp;
+		va_start(ap, argv0);
+		argv[0] = (char *)argv0;
+		for (i=1; i<=argc; i++)
+			argv[i] = va_arg(ap, char *);
+		envp = va_arg(ap, char **);
+		va_end(ap);
+		return _execle(path, argv, envp);
+	}
+}
+
+int execlp(const char* file, const char* argv0, ...) {
```

Comment:
Check out the [implementation in musl](https://git.musl-libc.org/cgit/musl/tree/src/process/execlp.c). You shouldn't assume that a box has `/bin/sh` here. This is more like an implementation of `system`.

Reply:
I have no idea what's going on in the musl implementation of execvp